spec/actions/model-spec.js

```diff
@@ -8,125 +8,254 @@ describe('model action creators', () => {
+          validateOptions,
+        }));
+
+        expect(ModelSpy).toHaveBeenCalledWith();
+        expect(validate).toHaveBeenCalledWith(attributes, validateOptions);
+      });
     });
   });
 
-  describe('destroy', () => {
-    beforeEach(function () {
-      this.destroy = modelActions.destroy;
+  describe('when an adapter is passed in', () => {
+    describe('fetch', () => {
+      beforeEach(function () {
+        this.fetch = modelActions.fetch;
+      });
+
+      it('calls fetch on the adapter', function () {
+        const deferred = $.Deferred(); // eslint-disable-line new-cap
+        const stubAdapter = {
+          fetchModel: jasmine.createSpy('fetchModel').and.returnValue(deferred),
+        };
+        const fetchOptions = { filters: { foo: 'bar' } };
+        this.store.dispatch(this.fetch('posts', '76', {
+          fetchOptions,
+          adapter: stubAdapter,
+        }));
+
+        expect(stubAdapter.fetchModel).toHaveBeenCalledWith('posts', '76', {
+          dispatch: jasmine.any(Function),
+          getState: jasmine.any(Function),
+          fetchOptions,
+        });
+      });
     });
 
-    it('sends destroy to the subscriber for the existing model', function () {
-      const posts = this.storageManager.storage('posts');
-      posts.add({ id: 1, title: 'What is redux?', message: 'I do not know but it might be awesome' });
-      const model = posts.last();
-      this.spy = spyOn(model, 'destroy').and.returnValue($.Deferred()); // eslint-disable-line new-cap
-      this.store.dispatch(this.destroy('posts', '1'));
+    describe('save', () => {
+      beforeEach(function () {
+        this.save = modelActions.save;
+      });
+
+      it('calls save on the adapter', function () {
+        const deferred = $.Deferred(); // eslint-disable-line new-cap
+        const stubAdapter = {
+          saveModel: jasmine.createSpy('saveModel').and.returnValue(deferred),
+        };
+        const saveOptions = { filters: { foo: 'bar' } };
+        const attributes = {};
+        this.store.dispatch(this.save('posts', '76', attributes, {
+          saveOptions,
+          adapter: stubAdapter,
+        }));
 
-      expect(this.spy).toHaveBeenCalled();
+        expect(stubAdapter.saveModel).toHaveBeenCalledWith('posts', '76', attributes, {
+          dispatch: jasmine.any(Function),
+          getState: jasmine.any(Function),
+          saveOptions,
+        });
+      });
     });
 
-    it('cancels previous requests with the same provided track key if they are pending', function () {
-      const xhrResultDouble = {
-        abort: jasmine.createSpy('abort'),
-        state: () => 'pending',
-      };
-      spyOn(this.storageManager.storage('posts').model.prototype, 'destroy').and.returnValue(xhrResultDouble);
+    describe('destroy', () => {
+      beforeEach(function () {
+        this.destroy = modelActions.destroy;
+      });
+
+      it('calls destroy on the adapter', () => {
```

Comment:
What is going on here.

Reply:
To reply to your past few comments re: "what is going on here", I'd suggest looking at the diff side by side as it will be easier to see that the existing specs were wrapped in an additional describe block. This causes the diff to appear a bit wonky, which you've likely seen before. I hope that clears things up!

Reply:
There are nested `it` specs in `it` specs. Can we fix that?

Reply:
Ah, I see what you mean. Fixed it!